pkg/runner/step_context.go

```diff
@@ -354,10 +359,35 @@ func (sc *StepContext) runAction(actionDir string, actionPath string) common.Exe
 				image = fmt.Sprintf("act-%s", strings.TrimLeft(image, "-"))
 				image = strings.ToLower(image)
 				contextDir := filepath.Join(actionDir, actionPath, action.Runs.Main)
+
+				exists, err := container.ImageExistsLocally(ctx, image, "any")
+				if err != nil {
+					return err
+				}
+
+				if exists {
+					wasRemoved, err := container.DeleteImage(ctx, image)
+					if err != nil {
+						return err
+					}
+					if !wasRemoved {
+						return fmt.Errorf("failed to delete image '%s'", image)
+					}
+				}
+
 				prepImage = container.NewDockerBuildExecutor(container.NewDockerBuildExecutorInput{
 					ContextDir: contextDir,
 					ImageTag:   image,
+					Platform:   rc.Config.ContainerArchitecture,
 				})
+				exists, err = container.ImageExistsLocally(ctx, image, rc.Config.ContainerArchitecture)
+				if err != nil {
+					return err
+				}
+
+				if !exists {
+					return err
+				}
```

Comment:
The code always exits here with err `nil` and the code below won't get executed, which causes local docker actions to be skipped.
This is caused by `container.NewDockerBuildExecutor` returning a function which is not executed here, but in line 483.
/cc @catthehacker @cplee